backupccl: deflake multinode datadriven tests that set cluster settings

msbutler opened the pull request:
The `SET CLUSTER SETTING` stmt propogrates the cluster setting to remote nodes asynchronously, but our data driven multinode tests incorrectly assumed that settings were propogated during stmt execution. This patch adds a new 'set-cluster-setting' dd cmd which will propogate the cluster setting to all nodes before the cmd returns.

Fixes #92808

Release note: none

msbutler committed:
backupccl: delake multinode datadriven tests that set cluster settings

The `SET CLUSTER SETTING` stmt propogrates the cluster setting to remote nodes
asynchronously, but our data driven multinode tests incorrectly assumed that
settings were propogated during stmt execution. This patch adds a new
'set-cluster-setting' dd cmd which will propogate the cluster setting to all
nodes before the cmd returns.

Fixes #92808

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/92911)
<!-- Reviewable:end -->

msbutler commented on pkg/ccl/backupccl/datadriven_test.go:
```diff
@@ -87,6 +87,10 @@ type sqlDBKey struct {
 
 type datadrivenTestState struct {
+	// clusters maps a name to its cluster
+	clusters map[string]serverutils.TestClusterInterface
```
I wonder if we ought to change the `new-server` cmd to `new-cluster`, as it's really spinning up a new cluster and pushing all sql cmds to get served by the first server in the test cluster.

adityamaru replied:
sgtm 👍

msbutler replied:
just pushed an extra commit for this.

msbutler commented:
nice sleuthing! for posterity, it looks like this bug may have been introduced by https://github.com/cockroachdb/cockroach/pull/89630

msbutler committed:
backupccl: rename server with cluster in dd driver

Previously, whenever the dd driver worked with a test cluster, it was always
called a server instead of a cluster. This is confusing, given all the other
ways crdb uses the word "server". To clear the air, this patch replaces all
uses of the word "server" with "cluster", where appropriate.

Epic: none

Release note: None

msbutler commented:
TFTR!

bors r=adityamaru
